### DISCUSSION TITLE
Not able to stream large videos using File View

### DISCUSSION THREAD

**Discussion by satty26** [appwrite/appwrite#5868] [Q&A]
I am not able to play videos larger than 1 GB. I am using the Chewie video player and using the **Get File View** API endpoint. I have uploaded the video on storage and it is mp4 type. The video is playing fine on the web but 

I am getting the following error on Android:

`Unhandled Exception: PlatformException(VideoError, Video player had error com.google.android.exoplayer2.ExoPlaybackException: Source error, null, null)
`

On IOS:

`Unhandled Exception: PlatformException(VideoError, Failed to load video: The request timed out., null, null)`

**Comment by jspm2013** [comment]
Hi Satty, I am sorry you did not get an answer here...did you manage to handle the errors, so that you could go on with you app on mobiles...?

**Comment by joern-h** [comment]
Pretty sure it is because the client sdk returns a ByteArray instead of an InputStream.
In android sdk, getFileView/Download returns only after the complete file is downloaded.

I had a similar issue while downloading big files and already raised an issue: https://github.com/appwrite/sdk-for-android/issues/44

I fixed it by patching the client sdk: https://github.com/joern-h/sdk-for-android/commit/60d10be7567dd67d119e30e252cea690a7cbcb96
Maybe this will help you.

Also, I just added a feature request for the sdk-generator that will hopefully fix our problems:
Maybe you could give it some visibility by voting for it ^^ https://github.com/appwrite/sdk-generator/issues/745
